crates/polars-ops/src/frame/join/mod.rs

```diff
@@ -418,11 +410,12 @@ pub trait DataFrameJoinOps: IntoDf {
     /// | 100             | null   |
     /// +-----------------+--------+
     /// ```
-    fn left_join<I, S>(&self, other: &DataFrame, left_on: I, right_on: I) -> PolarsResult<DataFrame>
-    where
-        I: IntoIterator<Item = S>,
-        S: Into<PlSmallStr>,
-    {
+    fn left_join(
+        &self,
+        other: &DataFrame,
+        left_on: impl IntoIterator<Item = impl Into<PlSmallStr>>,
+        right_on: impl IntoIterator<Item = impl Into<PlSmallStr>>,
```

Comment:
pre-work for removing `From<&&str>` - as then calling `join(left_on=["a"], right_on=["a", "b"])` becomes 2 distinct generics `<[&str; 1] as IntoIterator>`, `<[&str; 2] as IntoIterator>`